### ISSUE TITLE
Chrome Query Problem

### ISSUE THREAD

**Issue by amirzenoozi** [jaydenseric/apollo-upload-client#126]
Hello
When i Apollo Upload Client i will get error for all query and mutation in Chrome Browser but in Firefox everything is fine

I will get error with this code : 
```typescript
const httpLink = createUploadLink({
      uri: uri,
      withCredentials: true,
      method: 'POST',
    });
```

this code is work fine but i can't upload:
```typescript
const httpLink = httpLink.create({
      uri: uri,
      withCredentials: true,
      method: 'POST',
    });
```

Error: 
```
errors: [{message: "Must provide query string."}]
```

please help me

**Comment by jaydenseric** [comment]
Did you accidentally create this issue? There’s no description…

**Comment by amirzenoozi** [comment]
sorry..
description add

**Comment by jaydenseric** [comment]
I don't think `withCredentials` is a real option: https://github.com/jaydenseric/apollo-upload-client#function-createuploadlink

If there is a difference between Chrome and other browsers like Firefox, it is might be due to however you are approaching polyfills for APIs like `fetch`.

**Comment by jaydenseric** [comment]
Did you solve the problem in the end?

**Comment by jaydenseric** [comment]
Closing due to lack of activity. Feel free to start the conversation back up if the issue is not resolved :)

**Comment by strass** [comment]
I'm hitting the same `Must provide query string` error.

I've switched my apollo link to the one generated by createUploadLink. The payloads look as the following:
<img width="1331" alt="Screen Shot 2019-07-17 at 2 31 28 PM" src="https://user-images.githubusercontent.com/134242/61401065-9f74a900-a89f-11e9-8b38-90cf5de13e43.png">
<img width="1325" alt="Screen Shot 2019-07-17 at 2 31 37 PM" src="https://user-images.githubusercontent.com/134242/61401066-9f74a900-a89f-11e9-9e24-9932209e5cc5.png">

The response errors with: `{"errors":[{"message":"Must provide query string."}]}`

Regular mutations work fine, only file uploads fail with the above error. Do I need to do any manipulation of the data on the way out?

**Comment by mkgo** [comment]
I am having the same "Must provide query string" error. All other mutations work fine. The upload request is in Form Data. I am using the Hapi implementation of apollo server in the back and I am wondering if it might be the problem...

**Comment by DavidVrgs** [comment]
I am sorry to revive this topic, but someone found the solution?

**Comment by techlab23** [comment]
I'm also having same issue @jaydenseric.

**Comment by timohus** [comment]
I had the same issue and I was able to resolve it.
My setup is: React with Apollo in the front and Django with Graphene in the back.

As per the docs: "Also ensure the GraphQL server implements the GraphQL multipart request spec and that uploads are handled correctly in resolvers." In the backend, I'm using [graphene-file-upload](https://github.com/lmcgartland/graphene-file-upload). It turned out that while the mutation was configured correctly I forgot to configure the route.

Original route was: `path("graphql", GraphQLView.as_view(graphiql=True))`. Changed to: `path("graphql", FileUploadGraphQLView.as_view(graphiql=True))`.

`FileUploadGraphQLView` is provided by the `graphene-file-upload` package and implements exactly what the `apollo-upload-client` needs the backend to have (GraphQL multipart)

So my suggestion: check your backend, make sure it implements GraphQL multipart request spec
